src/main/thrift/parquet.thrift

```diff
@@ -660,6 +660,22 @@ struct ColumnMetaData {
   13: optional list<PageEncodingStats> encoding_stats;
 }
 
+struct ColumnCryptoMetaData {
+  /** Path in schema **/
+  1: required list<string> path_in_schema
```

Comment:
ColumnMetaData keeps sensitive information (stats etc) that has to be protected for secret columns.   Typically, these columns are encrypted with a key different from the footer key (if footer is encrypted at all). Hence, the writer encrypts the ColumnMetaData with the column key, stores it separately, and doesn't set the optional "meta_data" field in the ColumnChunk. Upon decryption, the reader needs to find the key for this column, possibly by the column name/path. But the name / path_in_schema is kept in ColumnMetadata - which is encrypted now. So this is duplicated in ColumnCryptoMetaData.

Reply:
We'll have to look into the design doc a bit deeper as it's not immediately obvious to us how this is exacly supposed to work, but based on your description this seems to be only needed when column key encryption is used, thereby it should be a field in the new struct `EncryptionWithColumnKey` we suggested.

Reply:
Correct. The metadata is not split in case of EncryptionWithFooterKey, since the footer encryption already protects it. Lets move it to the EncryptionWithColumnKey struct.